code/modules/surgery/organs/heart.dm

```diff
@@ -227,3 +227,232 @@
+/obj/item/organ/heart/ethereal/proc/on_take_damage(datum/source, damage, damagetype, def_zone)
+	SIGNAL_HANDLER
+	if(damagetype != BRUTE)
+		return
+
+	crystalization_process_damage += damage
+
+	if(crystalization_process_damage < BRUTE_DAMAGE_REQUIRED_TO_STOP_CRYSTALIZATION)
+		return
+
+	var/mob/living/carbon/human/ethereal = source
+
+	to_chat(ethereal, "<span class='userwarning'>The crystals on your body have completely broken</span>")
+	ethereal.visible_message("<span class='notice'>The crystals on [ethereal] are completely shattered and stopped growing</span>")
+
+	stop_crystalization_process(ethereal)
+
+/obj/structure/ethereal_crystal
+	name = "Ethereal Resurrection Crystal"
+	desc = "It seems to contain the corpse of an ethereal mending its wounds."
+	icon = 'icons/obj/ethereal_crystal.dmi'
+	icon_state = "ethereal_crystal"
+	max_integrity = 100
+	resistance_flags = FIRE_PROOF
+	density = TRUE
+	anchored = TRUE
+	///The organ this crystal belongs to
+	var/obj/item/organ/heart/ethereal/ethereal_heart
+	///Timer for the healing process. Stops if destroyed.
+	var/crystal_heal_timer
+	///Is the crystal still being built? True by default, gets changed after a timer.
+	var/being_built = TRUE
+
+/obj/structure/ethereal_crystal/Initialize(mapload, obj/item/organ/heart/ethereal/ethereal_heart)
+	. = ..()
+	src.ethereal_heart = ethereal_heart
+	ethereal_heart.owner.visible_message("<span class='notice'>The crystals fully encase [ethereal_heart.owner]!</span>")
+	to_chat(ethereal_heart.owner, "<span class='notice'>You are encased in a huge crystal!</span>")
+	playsound(get_turf(src), 'sound/effects/ethereal_crystalization.ogg', 50)
+	var/atom/movable/possible_chair = ethereal_heart.owner.buckled
+	possible_chair?.unbuckle_mob(ethereal_heart.owner, force = TRUE)
+	ethereal_heart.owner.forceMove(src) //put that ethereal in
+	add_atom_colour(ethereal_heart.ethereal_color, FIXED_COLOUR_PRIORITY)
+	crystal_heal_timer = addtimer(CALLBACK(src, .proc/heal_ethereal), CRYSTALIZE_HEAL_TIME, TIMER_STOPPABLE)
+	set_light(4, 10, ethereal_heart.ethereal_color)
+	update_icon()
+	flick("ethereal_crystal_forming", src)
+	addtimer(CALLBACK(src, .proc/start_crystalization), 1 SECONDS)
+
+/obj/structure/ethereal_crystal/proc/start_crystalization()
+	being_built = FALSE
+	update_icon()
+
+
+/obj/structure/ethereal_crystal/obj_destruction(damage_flag)
+	playsound(get_turf(ethereal_heart.owner), 'sound/effects/ethereal_revive_fail.ogg', 100)
+	return ..()
+
+
+/obj/structure/ethereal_crystal/Destroy()
+	if(!ethereal_heart)
+		return ..()
+	ethereal_heart.current_crystal = null
+	COOLDOWN_START(ethereal_heart, crystalize_cooldown, CRYSTALIZE_COOLDOWN_LENGTH)
+	ethereal_heart.owner.forceMove(get_turf(src))
+	REMOVE_TRAIT(ethereal_heart.owner, TRAIT_CORPSELOCKED, SPECIES_TRAIT)
+	deltimer(crystal_heal_timer)
+	visible_message("<span class='notice'>The crystals shatters, causing [ethereal_heart.owner] to fall out</span>")
+	return ..()
```

Comment:
This shouldn't have side effects, Destroy() should just clear references and then delete everything inside of it.
If you want to drop the ethereal from within, do that when the object is broken just before that calls qdel

Reply:
I have been unable to figure out where I could move this stuff to as so far every attempt I have done has just resulted in both the crystal and the player character getting deleted and a runtime error being produced.

Reply:
do it in obj_destruction and in heal_ethereal. If its qdeleted directly (from build mode or being destroyed not through obj_destruction, the ethereal inside should also be deleted)

Reply:
Righty that works...only got one new problem instead
`Runtime in garbage.dm, line 380: /obj/structure/ethereal_crystal destroy proc was called multiple times, likely due to a qdel loop in the Destroy logic`

Reply:
Ah does the heart getting deleted destroy the crystal too?

Reply:
Yeah, apparently deleting the player character deletes the crystal like…three times, each time resetting it to null

Reply:
Best I can tell removing the heart triggers a proc that qdel_nulls the crystal and then the removal proc itself also qdel_nulls the crystal. 

Then there’s yet another proc that fires off when the heart is deleted which triggers the same proc used by the surgery removal proc, so it also qdel_nulls the crystal. 

That proc was apparently added in to prevent players from being sent to the shadow realm upon getting deleted because they’re stuck in the body as a ghost

Reply:
This is a good indication that there is some janky code, however it could be somewhere in mob since the entire thing is basically jank.
Ideally the behaviour should be:
 - Deleting the crystal directly will delete the mob inside it
 - Deleting the mob inside the signal is handled somehow by the crystal (signal to either remove the reference, or the signal also triggers the mob deletion)
 - Deleting the heart should drop the mob.
 - Destroying the crystal normally should not delete the mob

The crystal stuff can be done in obj_destruction of the crystal since its the structure and the others should be done via the qdel signal rather than in /Destroy() for cleanliness (Otherwise we end up with messes like with modular PCs where they drop things on the ground when they are meant to be fully destroyed).